### ISSUE TITLE
Dealing with the rubyzip vulnerability...

### ISSUE THREAD

**Issue by duffyjp** [straydogstudio/axlsx_rails#77]
https://nvd.nist.gov/vuln/detail/CVE-2017-5946

If I force [rubyzip](https://github.com/rubyzip/rubyzip) to be at or above the minimum secure version, my axlsx_rails drops down to `0.1.5` and axlsx to `1.3.6`

```ruby
gem 'axlsx_rails'  
#gem 'axlsx', '>= 2.1.0.pre'  
gem 'rubyzip', '>= 1.2.1'    
```

```bash
bundle update axlsx_rails rubyzip axlsx
```

```diff
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -58,12 +58,12 @@ GEM
-    axlsx (2.1.0.pre)
+    axlsx (1.3.6)
-      rubyzip (~> 1.1.7)
-    axlsx_rails (0.4.0)
-      axlsx (>= 2.0.1)
+      rubyzip (>= 0.9.5)
+    axlsx_rails (0.1.5)
+      axlsx
-    rubyzip (1.1.7)
+    rubyzip (1.2.1)

```

[This Issue](https://github.com/randym/axlsx/issues/513) on [axlsx](https://github.com/randym/axlsx) states the required rubyzip version is updated on their master, but I don't see it in any published versions.  

I don't think this issue is in any way on your end, I just wanted to make sure you're aware of that CVE.  It sounds pretty nasty.

**Comment by duffyjp** [comment]
If anybody else needs to get this going ASAP, I can confirm forcing axlsx to the commit where the rubyzip version was changed works.

```ruby
gem 'axlsx_rails'
gem 'axlsx',
    git: 'https://github.com/randym/axlsx.git',
    ref: '776037c0fc799bb09da8c9ea47980bd3bf296874'
gem 'rubyzip', '>= 1.2.1'            # Rubyzip before 1.2.1 has a directory traversal vulnerability: CVE-2017-5946
```

You'll end up with:

```
axlsx_rails (0.5.1)
rubyzip (>= 1.2.1)
axlsx!
```

**Comment by straydogstudio** [comment]
Thanks for the update. I'll get this into the readme.

**Comment by straydogstudio** [comment]
I'm losing my memory. :P Finally got this in the README. 

@duffyjp Let me know if you've had any issues.

**Comment by btihen** [comment]
hello, are you considering updating the gem with this fix? 
That would be handy.

**Comment by straydogstudio** [comment]
I'll see what I can do. I've been reluctant to make the gem require 2.1.0.pre. I've not had time to work out all the possible dependency issues that would make.

**Comment by btihen** [comment]
Thanks,

The problem for me is I can include and the special config when I use it in a n app and run bundle install, but if I include it in a gem that depends on axlsx - then the gemspec file seems to get confused (& it only works if I use a bundle install on the gem and then locally install it - so so convenient for an automated install.

I am not sure I have enough experience to help with the possible issues that would create with your code, but if you tell me what I would need to do to check and help you - I could attempt to help.

Thanks,

Bill

**Comment by straydogstudio** [comment]
@btihen I finally got to poke around this. I'm as stuck as you since a released gem can only depend on released gems. So until Axlsx is released with a 1.2.1 dependency nothing can be done, and that release has been pending for a long time. (I've been wondering what to do with that.) Your gem users would have to require axlsx and point to the repo. I've initiated some communication to find out where Axlsx is going.
